dashboard/app/controllers/user_levels_controller.rb

```diff
@@ -29,7 +29,13 @@ def delete_predict_level_progress
     return head :forbidden, text: 'User must be instructor of course' unless script.can_be_instructor?(current_user)
     level = Level.find(params[:level_id])
     return head :not_found, text: 'Level not found' unless level
-    return head :bad_request, text: "Clearing progress on level type #{level.type} is not supported" unless ['Multi', 'FreeResponse'].include?(level.type)
+    is_allowed_level_type = ['Multi', 'FreeResponse'].include?(level.type)
+    # Some lab2 levels have a predict_settings field that indicates they are predict levels,
+    # rather than being a separate multi or free response level. We allow clearing progress on these levels.
+    if level.uses_lab2? && level.predict_settings && level.predict_settings["isPredictLevel"]
+      is_allowed_level_type = true
+    end
+    return head :bad_request, text: "Clearing progress on level type #{level.type} is not supported" unless is_allowed_level_type
```

Comment:
I think you could also do the following?
```
def delete_predict_level_progress
  ...

  unless clearable_level_type?(level)
    return head :bad_request, text: "Clearing progress on level type #{level.type} is not supported"
  end

  UserLevel.where(user_id: current_user.id, script_id: script.id, level: level.id).destroy_all
  head :ok
end

# Some lab2 levels have a predict_settings field that indicates they are predict levels. We allow clearing progress on these levels as well as separate multi or free-response levels.
private def clearable_level_type?(level)
  ['Multi', 'FreeResponse'].include?(level.type) || (level.uses_lab2? && level.predict_settings&.dig("isPredictLevel"))
end
```

Reply:
good idea!